### PR TITLE
Workaround for FTP servers, that does not support UTF8 option.

### PR DESCRIPTION
**Pull request by larrabee**

### CODE REVIEW

**Comment by rocheio** [comment]
+1 for this PR - we're encountering the same issue when trying to login to [NcFTPd servers](https://www.ncftp.com/ncftpd/) which show UTF-8 as an option in the FEAT listing but responds with `501 Option not recognized.` when running this OPTS command.

As it is, we are required to either catch and ignore this error via string comparison in all our code using this library, or keep this library pinned at the [Dec 31, 2016 version](https://github.com/jlaffaye/ftp/tree/f74506daaa13584329b9eb060d36e1cbd1fdd1d7) which was the last time that the UTF-8 command was not required for login.

We considered submitting a patch to the server itself to return a 202 code like `filezilla-server` does, but that server is no longer maintained / being updated, so we must catch it here

File: ftp.go [hunk 1]
```diff
@@ -184,6 +184,10 @@ func (c *ServerConn) setUTF8() error {
 	}
 
 	code, message, err := c.cmd(-1, "OPTS UTF8 ON")
+	// Workaround for FTP servers, that does not support this option.
+	if code == StatusBadArguments {
```
Comment (blaskovicz) [inline ftp.go]: I think this block should be after the next block, `if err != nil` since `c.cmd()` seems to return `0` as the code on error.

**Comment by blaskovicz** [comment]
Same deal here with NcFTPd.

```sh
$ telnet somehost 21
Connected to somehost.
Escape character is '^]'.
220 NcFTPd Server (licensed copy) ready.
OPTS UTF8 ON
501 Option not recognized.
```
I fetched the patch and verified that I was able to connect to the FTP server once this code was added.

👍 after my review comment.

File: ftp.go (hunk 1 above)
Comment (larrabee) [reply ftp.go]: You are right, done.

**Pull request merged**